src/scripts/components/VideoListItem.js

```diff
@@ -103,21 +103,15 @@ class UploadListItem extends Component<Props, void> {
       this.setState({ transcodingProgress: 100 })
     }
 
-    this.setState({
+    this.setState((prevState, nextProps) => ({
       totalProgress: Math.round(
-        (this.state.uploadProgress + this.state.transcodingProgress) / 2
+        (prevState.uploadProgress + prevState.transcodingProgress) / 2
```

Comment:
Looks like this is duplicated in two components. If extracted to a selector, then we have no problem. The transformed data can be passed to both components w/ ease.